src/coupled_driver.cpp

```diff
@@ -484,17 +498,25 @@ void CoupledDriver::init_heat_source()
 {
   comm_.message("Initializing heat source");
 
-  heat_source_ = xt::empty<double>({n_cells_});
-  heat_source_prev_ = xt::empty<double>({n_cells_});
+  const auto& neutronics = this->get_neutronics_driver();
+  const auto& heat = this->get_heat_driver();
+
+  if (neutronics.comm_.is_root() || heat.active()) {
+    heat_source_ = xt::empty<double>({n_cells_});
+    heat_source_prev_ = xt::empty<double>({n_cells_});
+  }
 }
 
 void CoupledDriver::set_heat_source()
 {
-  // Neutronics driver has heat source on each of its ranks. We need to make
+  const auto& neutronics = this->get_neutronics_driver();
+  auto& heat = this->get_heat_driver();
```

Comment:
`neutronics` is not used. And `heat` is not used until further down; I'd put its declaration back where it was before